threads/src/main/kotlin/pl/edu/pw/ia/threads/domain/command/aggregate/Moderator.kt

```diff
@@ -0,0 +1,48 @@
+package pl.edu.pw.ia.threads.domain.command.aggregate
+
+import java.util.UUID
+import org.axonframework.commandhandling.CommandHandler
+import org.axonframework.modelling.command.AggregateIdentifier
+import org.axonframework.modelling.command.AggregateLifecycle
+import org.axonframework.spring.stereotype.Aggregate
+import pl.edu.pw.ia.shared.domain.command.AddModeratorCommand
+import pl.edu.pw.ia.shared.domain.command.RemoveModeratorCommand
+import pl.edu.pw.ia.shared.domain.event.ModeratorAddedEvent
+import pl.edu.pw.ia.shared.domain.event.ModeratorRemovedEvent
+
+@Aggregate
+internal class Moderator {
+
+	@AggregateIdentifier
+	private lateinit var moderatorId: UUID
+	private lateinit var accountId: UUID
+	private lateinit var threadId: UUID
+
+	private constructor()
+
+	@CommandHandler
+	constructor(command: AddModeratorCommand) {
+		// TODO: Check if command.accountId is moderator in command.threadId
+		AggregateLifecycle.apply(
+			ModeratorAddedEvent(
+				moderatorId = command.moderatorId,
+				threadId = command.threadId,
+				accountId = command.subjectAccountId
+			)
+		)
+	}
+
+	@CommandHandler
+	fun handle(command: RemoveModeratorCommand) {
+		// TODO: Check if command.accountId is moderator in command.threadId
+		// TODO: Should we allow moderator to delete himself from moderators?
```

Comment:
hmm, we shouldn't allow to delete only the original moderator - aka post creator 🤔

Reply:
Let's assume only author of the thread can modify moderators.